cloud/openstack/os_project.py

```diff
@@ -0,0 +1,201 @@
+#!/usr/bin/python
+# Copyright (c) 2015 IBM Corporation
+#
+# This module is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This software is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this software.  If not, see <http://www.gnu.org/licenses/>.
+
+
+try:
+    import shade
+    HAS_SHADE = True
+except ImportError:
+    HAS_SHADE = False
+
+DOCUMENTATION = '''
+---
+module: os_project
+short_description: Manage OpenStack Projects
+extends_documentation_fragment: openstack
+version_added: "2.0"
+author: "Alberto Gireud (@agireud)"
+description:
+    - Manage OpenStack Projects. Projects can be created,
+      updated or deleted using this module. A project will be updated
+      if I(name) matches an existing project and I(state) is present.
+      The value for I(name) cannot be updated without deleting and
+      re-creating the project.
+options:
+   name:
+     description:
+        - Name for the project
+     required: true
+   description:
+     description:
+        - Description for the project
+     required: false
+     default: None
+   domain_id:
+     description:
+        - Domain id to create the project in if the cloud supports domains
+     required: false
+     default: None
+   enabled:
+     description:
+        - Is the project enabled
+     required: false
+     default: True
+   state:
+     description:
+       - Should the resource be present or absent.
+     choices: [present, absent]
+     default: present
+requirements:
+    - "python >= 2.6"
+    - "shade"
+'''
+
+EXAMPLES = '''
+# Create a project
+- os_project:
+    cloud: mycloud
+    state: present
+    name: demoproject
+    description: demodescription
+    domain_id: demoid
+    enabled: True
+
+# Delete a project
+- os_project:
+    cloud: mycloud
+    state: absent
+    name: demoproject
+'''
+
+
+RETURN = '''
+project:
+    description: Dictionary describing the project.
+    returned: On success when I(state) is 'present'
+    type: dictionary
+    contains:
+        description:
+            description: Project description
+            type: string
+            sample: "demodescription"
+        domain_id:
+            description: Project domain ID. Only present with Keystone >= v3.
+            type: string
+            sample: "default"
```

Comment:
I intended it to be Domain ID (uuid) but it looks like I might have been mistaken in including it based off of https://github.com/openstack/python-keystoneclient/blob/master/keystoneclient/v3/projects.py#L26.